docs/source/rfc-configuration-overhaul.md

```diff
@@ -0,0 +1,893 @@
+Here's a snippet from the current configuration showcasing this:
+
+```json
+{
+  "PUBLIC_KEY": null,
+  "PRIVATE_KEY": null,
+  "DISABLE_PANIC_TERMINAL_COLORS": false
+}
+```
+
+---
+
+In summary, these issues underscore the need for clearer, more intuitive naming conventions in Iroha 2's configuration.
+
+### Rustisms in the Configuration Reference
+
+Configuration reference contains Rust-specific expressions that wouldn't make sense to the end-user who is configuring
+Iroha and might not be familiar with Rust. Here are some examples:
+
+- [Using `Option<..>`](https://github.com/hyperledger/iroha/blob/35ba182e1d1b6b594712cf63bf448a2edefcf2cd/docs/source/references/config.md#option)
+- [Using `std::path::PathBuf`](https://github.com/hyperledger/iroha/blob/35ba182e1d1b6b594712cf63bf448a2edefcf2cd/docs/source/references/config.md#loggerlog_file_path)
+
+The use of rustisms like that overcomplicates the configuration reference and affects the user experience. Further, it
+doesn't actually provide any extra information to a user familiar with Rust, as environment variables and JSON values
+cannot be `None` due to being stringly typed and null-able regardless of whether the type specifies `Option`.
+
+### Unhelpful Error Messages
+
+Some of the error messages related to Iroha configuration are inconsistent and misleading. Here's a quick rundown of
+various scenarios, associated errors and possible confusions.
+
+#### 1. No configuration file
+
+Running Iroha without a configuration file results in the following log and error:
+
+```
+IROHA_TORII: environment variable not found
+IROHA_SUMERAGI: environment variable not found
+IROHA_KURA: environment variable not found
+IROHA_BLOCK_SYNC: environment variable not found
+IROHA_PUBLIC_KEY: environment variable not found
+IROHA_PRIVATE_KEY: environment variable not found
+IROHA_GENESIS: environment variable not found
+Configuration file not found. Using environment variables as fallback.
+Error:
+   0: Please add `PUBLIC_KEY and PRIVATE_KEY` to the configuration.
+```
+
+Several things are wrong with this error message:
+
+- The information about using environment variables as fallback comes after the messages about them not being found.
+- Hinting to add `PUBLIC_KEY` and `PRIVATE_KEY` to the configuration does not explain what happened (the absence of the
+  config file and no environment variables to fallback to). This hint is also not helpful as there is no information on
+  whether public and private keys should be added to the config file or to ENV variables, and no information about the
+  ENV variables these fields are mapped to.
+
+#### 2. Path to non-existent config file
+
+Providing a path to a configuration file that doesn't exist results in the exact same error as if there was no
+configuration file specified at all.
+
+While it makes sense to silently fallback to ENV variables if the user doesn't provide a path to a config file, in case
```

Comment:
I guess ENV has the most priority for a parameter so it's rather not a fallback, though I get the sense anyway

Reply:
I see, let me try to change the phrasing.

Reply:
Rephrased.